File: src/services/ocr.ts
import { invoke } from '@tauri-apps/api/core';
import type { OcrResult } from './types';

export async function performOcr(imagePath: string, language?: string): Promise<OcrResult> {
  return await invoke<OcrResult>('perform_ocr', { imagePath, language });
}

export async function checkOcrTools(): Promise<string[]> {
  return await invoke<string[]>('check_ocr_tools');
}

File: src-tauri/src/commands/ocr.rs
use crate::ocr::create_engine;
use crate::types::OcrResult;

#[tauri::command]
pub fn perform_ocr(image_path: String, language: Option<String>) -> Result<OcrResult, String> {
    let engine = create_engine()?;
    let lang = language.unwrap_or_else(|| "eng".to_string());
    engine
        .perform_ocr(&image_path, &lang)
        .map_err(|e| e.to_string())
}

#[tauri::command]
pub fn check_ocr_tools() -> Vec<String> {
    crate::ocr::check_tools()
}
